data/committee/credential.go

```diff
@@ -155,11 +156,62 @@ func (cred Credential) Selected() bool {
 	return cred.Weight > 0
 }
 
+// lowestOutput is used for breaking ties when there are multiple proposals.
+// People will vote for the proposal whose credential has the lowest lowestOutput.
+//
+// We hash the credential and interpret the output as a bigint.
+// For credentials with weight w > 1, we hash the credential w times (with
+// different counter values) and use the lowest output.
+//
+// This is because a weight w credential is simulating being selected to be on the
+// leader committee w times, so each of the w proposals would have a different hash,
+// and the lowest would win.
 func (cred Credential) lowestOutput() *big.Int {
 	var lowest big.Int
 
 	h1 := cred.VrfOut
-	for i := uint64(0); i < cred.Weight; i++ {
+	// It is important that i start at 1 rather than 0 because cred.Hashable
+	// was already hashed with iter = 0 earlier (in UnauthenticatedCredential.Verify)
+	// for determining the weight of the credential. A nonzero iter provides
+	// domain separation between lowestOutput and UnauthenticatedCredential.Verify
+	//
+	// It we reused the iter = 0 hash output here it would be nonuniformly
+	// distributed (because lowestOutput can only get called if weight > 0).
+	// In particular if i starts at 0 then weight-1 credentials are at a
+	// significant disadvantage because UnauthenticatedCredential.Verify
+	// wants the hash to be large but tiebreaking between proposals wants
+	// the hash to be small.
+	for i := uint64(1); i <= cred.Weight; i++ {
```

Comment:
nit: for loop could be written better in terms of performance

Reply:
In practice cred.Weight is never above 6 or 7, so performance isn't really an issue here -- computing the VRF is going to be way more expensive than computing the lowest output